codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ConstrainedShapeSymbolProvider.kt

```diff
@@ -86,11 +87,9 @@ class ConstrainedShapeSymbolProvider(
                 }
             }
             is CollectionShape -> {
-                // TODO(https://github.com/awslabs/smithy-rs/issues/1401) Both arms return the same because we haven't
-                //  implemented any constraint trait on collection shapes yet.
                 if (shape.isDirectlyConstrained(base)) {
-                    val inner = this.toSymbol(shape.member)
-                    symbolBuilder(shape, RustType.Vec(inner.rustType())).addReference(inner).build()
+                    assert(constrainedCollectionCheck(shape)) { "Only the `length` constraint trait can be applied to lists" }
```

Comment:
My thinking was that we could start running expensive checks in `assert`s because we can disable them if they become too slow, but this is probably a larger conversation (and looking back, I doubt our checks will ever become a bottleneck).
I've changed this to use `check` in 3b92790a478a8d62fda86dc6492529af6cffc2f0.